.github/workflows/test.yml

```diff
@@ -38,11 +39,29 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip setuptools wheel
-          pip install -e .[dev,cfg,z3]
+          pip install -e .[dev,cfg]
+          if [ "${{ matrix.dependency }}" == "z3-solver" ]; then
+            pip install z3-solver
```

Comment:
Use `.[z3]` instead of `z3-solver` (this will ensure we're installing everything in the "z3" group, which may be added to in the future)